department-of-veterans-affairs/caseflow
# BillyS/APPEALS-7452: Creation of Completed tab in the CSP organization queue

Resolves [APPEALS-7452](https://vajira.max.gov/browse/APPEALS-7452)

### Description
A new completed tab was established to display cases that have a VhaDocumentSearchTask, Status of Completed and are less than 7 days old. As a CSP user, I need Caregiver cases to be routed to the Completed tab so that I know which cases have been sent back to Board Intake

### Acceptance Criteria
- [ ] Creation of the Completed tab in the CSP organization queue
- [ ] The following informational text appears under the Completed tab
   - [ ] Cases completed (last 7 days):
- [ ] The Completed tab contains the following columns
   - [ ] Badges
   - [ ] Case Details
   - [ ] Assigned By
   - [ ] Docket
   - [ ] Days Waiting
   - [ ] Veteran Documents

### Testing Plan
1. Open Caseflow App
2. Switch user to one of the new Caregiver users
![image](https://user-images.githubusercontent.com/98414141/182457704-7b5dbe54-8745-489b-bb46-c5964ad0b2a5.png)
3. Select Your Queue

![image](https://user-images.githubusercontent.com/98414141/182458385-e6e710b3-f3b8-41ed-b8f6-e090de6e481d.png)

4. Switch to the CSP organization's queue:
<img width="852" alt="switch-to-csp-queue" src="https://user-images.githubusercontent.com/99351305/181598516-51ff9792-ba56-4bcd-b9fd-e8aa3776954e.PNG">

5. You will end up here with the completed tab being the only visible tab so far:
     ![image](https://user-images.githubusercontent.com/98414141/182463995-f8a560ac-f498-4423-b8fb-ac8e68a959dd.png)
     a. Verify that the following text is visible on the screen "Cases completed (last 7 days):"
     b. Also verify that the following columns are present:

         Case Details | Tasks | Assigned By | Types | Docket | Days Waiting | Veteran Documents

```ruby
6. Open the Rails console (`make c` in VSCode Terminal)

7. Create appeal with a Pre-Docket Task
     appeal = FactoryBot.create(:appeal, :with_pre_docket_task)

8. Check task tree to make sure it created a root task and Pre-docket Task
    appeal.treee

    should display:
```
   ![image](https://user-images.githubusercontent.com/98414141/182461698-60c4200d-7d46-4037-bf40-80c126193936.png)
```ruby 
    NOTE the ID on the PreDocketTask for future use.  The example displayed is 7134
 
9. Define the task by the Task ID
    task=Task.find(7134)

10. Create the Vha Document Search Task
      vha_document_search_task=VhaDocumentSearchTask.create!( parent: task, appeal: appeal, assigned_at: Time.zone.now, 
      assigned_to: VhaCaregiverSupport.singleton )

11. Check task tree to make sure it created a Vha Docuement Search Task
      appeal.treee

     should display:
```
   ![image](https://user-images.githubusercontent.com/98414141/182461939-10a4fdea-c6d9-4882-9a01-f0e23b9ef062.png)

      This time, please NOTE the ID on the VhaDocketSearchTask for future use.  The example displayed is 7135. In addition, note 
       that the status on the VhaDocumentSearchTask is "assigned"
  
12. Go back to Caseflow and refresh the page. It should remain unchanged from step 5 because the status of the 
      created appeal is "assigned" rather than "completed"
```ruby
13. Return to Rails Console and change the VhaDocumentSearchTask Status to "completed"
      Task.find(7135).completed!
```
14. Go back to Caseflow and refresh the page.  Now the case should be visible in the completed tab
      ![image](https://user-images.githubusercontent.com/98414141/182463733-6f2a28e4-17d3-4991-8175-2acf5899b41a.png)
```ruby
15. Go back to Rails Console to test that an appeal more than 7 days old does not appear in the tab even when completed. 
      Task.find(7135).update(closed_at: 8.days.before) 
```
16. Go back to Caseflow and refresh the page.  You should observe that the appeal is no longer visible
```ruby
17. Go back to Rails Console to test that an appeal just less than 7 days old does not appear in the tab even when completed. 
      Task.find(7135).update(closed_at: 6.days.before)
```
18.  Go back to Caseflow and refresh the page.  You should observe that the appeal is visible once again as in step 14
```ruby
19. Follow step 13 to verify that any status other than complete results in the appeal not being displayed in the tab, verifying 
     the results in caseflow 
    
    Task.find(7135).on_hold!
    Task.find(7135).cancelled!
    Task.find(7135).in_progress!
